### ISSUE TITLE
constructor doesn't check finiteness for small maxE

### ISSUE THREAD

**Issue by tonyofbyteball** [MikeMcl/decimal.js#121]
I don't think anybody is using such small maxE but looks like there is a bug:
```
Decimal.set({maxE: 2, minE: -2});
Decimal(1e3).isFinite()     // true, expected false
Decimal(1e3).times(1).isFinite() // false, correct
```
It works fine for large maxE or when constructing from a string.

**Comment by MikeMcl** [comment]
Yes, it's a bug. Looks like *bignumber.js* has this issue also. 

Thanks for the report.

Notes:
Decimal fails to return Infinity when passed JS number integers less than `1e7`, when `maxE` is less than `7`. Bug introduced by 'fast path for small integers' constructor refactoring, *ba8cae8fc*, 12/10/2015, *v4.0.3*. I note that `maxE` is also not being checked when constructor is passed a Decimal instance to copy. Changing `maxE` obviously doesn't affect the value of Decimal instances already created, and the `precision` setting doesn't affect the value returned by the constructor, so I'm inclined to remove the checks from the constructor completely. I will consider this further.

**Comment by MikeMcl** [comment]
Fixed in *v10.1.0*.
